### PR TITLE
Refactor Rider warnings and hints:

### PR DESCRIPTION
**Pull request by zuev93**
- Add comments to disable unused warnings.
- Update access modifiers where needed.
- Extracted event invocation in a separate method where is needed.
- Merged multiple patches in BuildEvents.
- Replaced usage of EventHandler with Action where is feasible.
- Minor tweaks.

Feel free to ask to revert a certain changes.

### CODE REVIEW

File: MultiplayerMod/Multiplayer/Objects/Reference/GridReference.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public class GridReference : GameObjectReference {
     public GridReference(GameObject gameObject) {
         var extension = gameObject.GetComponent<GameObjectExtension>();
         Cell = Grid.PosToCell(gameObject);
-        Layer = extension?.GridLayer ?? 0;
+        Layer = extension != null ? extension.GridLayer : 0;
```
Comment (polycone) [inline MultiplayerMod/Multiplayer/Objects/Reference/GridReference.cs:21]: Not sure if it's required for our operations.

File: MultiplayerMod/Platform/Steam/Network/SteamClient.cs [hunk 2]
```diff
@@ -121,7 +121,7 @@ public class SteamClient : IMultiplayerClient {
         SteamFriends.SetRichPresence("connect", $"+connect_lobby {lobby.Id}");
     }
 
-    private SteamNetworkingIdentity GetNetworkingIdentity(CSteamID steamId) {
+    private static SteamNetworkingIdentity GetNetworkingIdentity(CSteamID steamId) {
```
Comment (polycone) [inline MultiplayerMod/Platform/Steam/Network/SteamClient.cs:124]: I propose not to use static in places where it's not supposed to be.

File: MultiplayerMod/Multiplayer/Objects/Reference/GridReference.cs (hunk 1 above)
Comment (zuev93) [reply MultiplayerMod/Multiplayer/Objects/Reference/GridReference.cs:21]: Done

File: MultiplayerMod/Platform/Steam/Network/SteamClient.cs (hunk 2 above)
Comment (zuev93) [reply MultiplayerMod/Platform/Steam/Network/SteamClient.cs:124]: Done

File: MultiplayerMod/Game/Mechanics/Objects/TargetExtractor.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 namespace MultiplayerMod.Game.Mechanics.Objects;
 
-static class TargetExtractor {
+internal static class TargetExtractor {
```
[the remark this answers is not in the document]
Comment (zuev93) [reply MultiplayerMod/Game/Mechanics/Objects/TargetExtractor.cs:9]: Done

**Pull request merged**